docs/asciidoc/http-client.adoc

```diff
@@ -512,3 +518,188 @@ include::{examplesdir}/resolver/custom/Application.java[lines=18..38]
 ----
 <1> Sets the JVM built-in resolver.
 ====
+
+== Timeout Configuration
+This section describes various timeout configuration options that can be used in `HttpClient`.
+Configuring a proper timeout may improve or solve issues in the communication process.
+The configuration options can be grouped as follows:
+
+* <<connection-pool-timeout>>
+* <<http-client-timeout>>
+** <<response-timeout>>
+** <<connection-timeout>>
+** <<ssl-tls-timeout>>
+** <<proxy-timeout>>
+** <<dns-timeout>>
+
+[[connection-pool-timeout]]
+=== Connection Pool Timeout
+By default, `HttpClient` uses a connection pool. When a request is completed successfully and if the connection is not scheduled for closing,
+the connection is returned to the connection pool and thus it can be reused for processing another request. The connection may
+be reused immediately for another request or may stay idle in the connection pool for some time.
+
+The below shows the available timeout configuration options:
+
+* `maxIdleTime` - the maximum time (resolution: ms) that this connection stays idle in the connection pool. By default,
+{javadoc}/reactor/netty/resources/ConnectionProvider.ConnectionPoolSpec.html#maxIdleTime-java.time.Duration-[maxIdleTime] is not specified.
+
+NOTE: When you configure `maxIdleTime`, you should consider the idle timeout configuration on the target server.
+Choose a configuration equal or less than the one on the target server thus you will be able to reduce the I/O
+issues caused by a connection closed by the target server.
+
+* `maxLifeTime` - the maximum time (resolution: ms) that this connection stays alive. By default,
+{javadoc}/reactor/netty/resources/ConnectionProvider.ConnectionPoolSpec.html#maxLifeTime-java.time.Duration-[maxLifeTime] is not specified.
+* `pendingAcquireTimeout` - the maximum time (resolution: ms) after which a pending acquire operation must complete, or
+a `PoolAcquireTimeoutException` will be thrown. Default: 45s.
+
+By default, these timeouts are checked on connection `release`/`acquire` operation and if some timeout is reached, the connection is closed and removed from the connection pool.
```

Comment:
Change to:

By default, these timeouts are checked on connection `release` or `acquire` operations and, if some timeout is reached, the connection is closed and